packages/video_player_videohole/tizen/src/drm_licence.cc

```diff
@@ -0,0 +1,767 @@
+  SDynamicBuf header;  // response header
+  SDynamicBuf body;    // response body
+  long resCode;
+};
+
+// Internal Static Functions
+static size_t ReceiveHeader(void* ptr, size_t size, size_t nmemb, void* stream);
+static size_t ReceiveBody(void* ptr, size_t size, size_t nmemb, void* stream);
+static size_t SendBody(void* ptr, size_t size, size_t nmemb, void* stream);
+static bool AppendData(SDynamicBuf* aBuf, const void* aData, size_t aSize);
+static char* GetRedirectLocation(const char* aHeaders, bool bSupportHttps);
+static struct curl_slist* CurlSlistAppend(struct curl_slist* aList,
+                                          const char* aString);
+static DRM_RESULT ComposePostData_TZ(SHttpSession* hSession,
+                                     const char* f_pbPostData, int f_cbPostData,
+                                     const char* f_extSoapHeader);
+static struct curl_slist* SetHttpHeader(CURL* pCurl,
+                                        CBmsDrmLicenseHelper::EDrmType f_type,
+                                        const char* f_pCookie,
+                                        const char* f_pHttpHeader,
+                                        const char* f_pUserAgent);
+static SHttpSession* HttpOpen(void);
+static int CbCurlProgress(void* ptr, double TotalToDownload,
+                          double NowDownloaded, double TotalToUpload,
+                          double NowUploaded);
+static DRM_RESULT HttpStartTransaction(
+    SHttpSession* hSession, const char* f_pUrl, const void* f_pbPostData,
+    unsigned f_cbPostData, CBmsDrmLicenseHelper::EDrmType f_type,
+    const char* f_pCookie, const char* f_pSoapHeader, const char* f_pHttpHeader,
+    const char* f_pUserAgent, bool* pCancelRequest);
+static void HttpClose(SHttpSession* hSession);
+
+bool AppendData(SDynamicBuf* aBuf, const void* aData, size_t aSize) {
+  size_t newSize = aBuf->iSize + aSize;
+  if (aBuf->iAllocated < newSize) {
+    newSize += 1024;
+    unsigned char* buf = (unsigned char*)realloc(aBuf->iData, newSize);
+    if (!buf) {
+      LOG_ERROR("[DrmLicence] AppendData : realloc fail ");
+      return false;
+    }
+    aBuf->iData = buf;
+    aBuf->iAllocated = newSize;
+    LOG_DEBUG(
+        "[DrmLicence] AppendData : realloc aSize(%d), iSize(%d) "
+        "aBuf->iAllocated(%d)",
+        aSize, aBuf->iSize, aBuf->iAllocated);
+  }
+  memcpy(aBuf->iData + aBuf->iSize, aData, aSize);
+  aBuf->iSize += aSize;
+
+  return true;
+}
+
+char* GetRedirectLocation(const char* aHeaders, bool bSupportHttps) {
+  if (!aHeaders) {
+    return nullptr;
+  }
+
+  const char* pLocation = strcasestr(aHeaders, "Location");
+  if (!pLocation) {
+    return nullptr;
+  }
+
+  const char* ptr = pLocation + strlen("Location");
+
+  while (*ptr == ':') {
+    ptr++;
+  }
+  while (*ptr == ' ') {
+    ptr++;
+  }
+
+  unsigned i = 0;
+  while (ptr[i] && (ptr[i] != ' ') && (ptr[i] != '\n') && (ptr[i] != '\r')) {
+    i++;
+  }
```

Comment:
How about adding an example of a header that knows why the following parts are needed?

Reply:
> How about adding an example of a header that knows why the following parts are needed?

Fixed.